crux-core/src/crux/kv/index_store.clj

```diff
@@ -452,22 +460,77 @@
 (defn- value-buffer->id-buffer [index-snapshot ^DirectBuffer value-buffer]
   (c/->id-buffer (db/decode-value index-snapshot value-buffer)))
 
+(defn- ecav-cache-lookup ^java.util.Map [ecav-cache ecav-i ^DirectBuffer eid-value-buffer ^DirectBuffer content-hash-buffer]
+  (lru/compute-if-absent ecav-cache
+                         content-hash-buffer
+                         mem/copy-to-unpooled-buffer
+                         (fn [content-hash-buffer]
+                           (let [eid-size (mem/capacity eid-value-buffer)
+                                 a->vs (HashMap.)
+                                 prefix (encode-ecav-key-to (.get seek-buffer-tl)
+                                                              eid-value-buffer
+                                                              content-hash-buffer)
+                                 i (new-prefix-kv-iterator ecav-i prefix)]
+                             (loop [k (kv/seek i prefix)]
```

Comment:
think this is `all-keys-in-prefix`?

Reply:
It is, but this is slightly faster, as there's no immediate sequence.

Reply:
Ah, right - in that case, should we apply these changes to `all-keys-in-prefix`?